### PR TITLE
MTG-404 Skip batch mint indexing feature

### PR DESCRIPTION
**Pull request by n00m4d**
# What
This PR extends/improves previously added feature - skip batch minted trees indexing.
# Why
Code with previous version was not indexing batch mint instructions, if specific flag was set, but if there were any changes in a tree after it created like mint, transfer and so on all that updates were indexed. It means that API would try to return proofs for that assets but it would be incorrect because most of the updates were missed such as initiall instruction was not parsed. Now indexer checks if merkle tree update we received is related to batched minted tree and if so it will not save its update, of course if skip indexing flag is set to true.
# How
There is hashSet added to the program transformer. On start that hashSet is populated with all the batched minted trees from the batch_mint_to_verify table and then it's updating during its work. When we get cNFT update we check if tree ID is in hashSet and if so - skip this update.

### CODE REVIEW

**Review by snorochevskiy — APPROVED** [review]

**Review by StanChe — COMMENTED** [review]
Several questions to clarify:
1. entries in batch_mint_to_verify are never dropped, right?
2. account_worker method has some hardcoded false in the list of arguments sent to ProgramTransformer::new. Is this intended?
3. the batch_mint_to_verify entries are always created, without regards to the value from config, am I getting it right?

File: nft_ingester/src/account_updates.rs [hunk 1]
```diff
@@ -35,6 +35,7 @@ pub fn account_worker<T: Messenger>(
                 pool,
                 create_download_metadata_notifier(bg_task_sender),
                 false,
+                false,
```
[the remark this answers is not in the document]
Comment (n00m4d) [reply nft_ingester/src/account_updates.rs:38]: Here account_worker is created and both `cl_audits` and `skip_batch_minted_trees` flags don't make any sense. During accounts handling they are not used

**Review by n00m4d — COMMENTED** [review]
1. right, there only status is updated
2. intended, because for account_worker both `cl_audits` and `skip_batch_minted_trees` flags don't make any sense. During accounts handling they are not used
3. yes because we create entry there during `FinalizeTreeWithRoot` instruction processing and that instruction will be processed even if we set `skip_batch_minted_trees` to true because at the moment indexer receive instruction it doesn't have tree ID in a hashSet yet

File: nft_ingester/src/account_updates.rs [hunk 2]
```diff
@@ -35,6 +35,7 @@ pub fn account_worker<T: Messenger>(
                 pool,
                 create_download_metadata_notifier(bg_task_sender),
                 false,
```
[the remark this answers is not in the document]
Comment (n00m4d) [reply nft_ingester/src/account_updates.rs]: not sure that we have to pass value from the config for account worker such as this flag is related to transaction processing only

File: nft_ingester/src/account_updates.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (StanChe) [reply nft_ingester/src/account_updates.rs]: I'd put that in a comment, as it's not crystal clear

**Pull request merged**